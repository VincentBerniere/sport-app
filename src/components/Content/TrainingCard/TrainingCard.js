import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import './TrainingCard.scss';

class TrainingCard extends Component {
    state = {
        redirect: false,
        description: ''
    };

    componentDidMount() {
        const maxSizeOfDescription = 200;

        if (this.props.training.description.length > maxSizeOfDescription) {
            this.setState({ description: this.props.training.description.substring(1, maxSizeOfDescription) + '...' });
        } else {
            this.setState({ description: this.props.training.description });
        }
    }

    goToTrainingDetail = () => {
        this.setState({ redirect: true });
    }

    renderRedirectToTrainingDetail = () => {
        if (this.state.redirect) {
            return <Redirect to={`/trainings/${this.props.training.id}/detail`} />
        }
    }

    render() {
        return ( 
            <div className="training-card ui card">
                <div className="content" onClick={this.goToTrainingDetail}>
                    <div className="header">
                        {this.props.training.title}
                    </div>

                    <div className="description">
                        {this.state.description}
                    </div>
                </div>
                    
                <div className="extra content">
                    <span className="left floated like">
                        <i className="like icon"></i>
                        Like
                    </span>
                    <span className="right floated star">
                        <i className="star icon"></i>
                        Favorite
                    </span>
                </div>

                {this.renderRedirectToTrainingDetail()}
            </div>
        );
    }
}

TrainingCard.defaultProps = {
    training: {}
}

export default TrainingCard;
