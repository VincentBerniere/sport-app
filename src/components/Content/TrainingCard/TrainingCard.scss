.training-card {
    >.content:first-child {
        cursor: pointer;
    }
}

.ui.cards > .card, .ui.card {
    width: 250px;
}