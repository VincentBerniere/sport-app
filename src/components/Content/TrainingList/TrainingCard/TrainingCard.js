import React, { Component } from 'react';

class TrainingCard extends Component {
    render() {
        return ( 
            <div className="ui card">
                <div className="content">
                    <i className="right floated like icon"></i>
                    <i className="right floated star icon"></i>

                    <div className="header">{this.props.training.title}</div>

                    <div className="description">
                        <p>
                        Comment prendre de la masse musculaire ? Les charges à déplacer et les rythmes d'exécution sont différents selon le but visé (esthétique, puissance, maintien d'un métabolisme basal élevé)
                        </p>
                    </div>
                </div>
                    
                <div className="extra content">
                    <span className="left floated like">
                        <i className="like icon"></i>
                        Like
                    </span>
                    <span className="right floated star">
                        <i className="star icon"></i>
                        Favorite
                    </span>
                </div>
            </div>
        );
    }
}

export default TrainingCard;