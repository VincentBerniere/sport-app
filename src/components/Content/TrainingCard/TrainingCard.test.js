import React from 'react';
import { shallow } from 'enzyme'; 
import TrainingCard from './TrainingCard';

describe('TrainingCard', () => {
    const trainingCard = shallow(<TrainingCard />);

    it('renders correctly', () => {
        expect(trainingCard).toMatchSnapshot();
    });

    it('displays the training from props', () => {
        const training = { id: 0, name: 'Muscu', author: 'Jack' };
        const trainingCard = shallow(<TrainingCard training={training} />);
        
        expect(trainingCard.find('.header').text()).toEqual(training.name);
    });
});