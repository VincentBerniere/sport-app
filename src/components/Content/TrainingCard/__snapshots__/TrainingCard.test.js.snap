// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TrainingCard
    training={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="content"
          onClick={[Function]}
        >
          <div
            className="header"
          />
          <div
            className="description"
          >
            
          </div>
        </div>,
        <div
          className="extra content"
        >
          <span
            className="left floated like"
          >
            <i
              className="like icon"
            />
            Like
          </span>
          <span
            className="right floated star"
          >
            <i
              className="star icon"
            />
            Favorite
          </span>
        </div>,
        undefined,
      ],
      "className": "training-card ui card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header"
            />,
            <div
              className="description"
            >
              
            </div>,
          ],
          "className": "content",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "header",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "description",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="left floated like"
            >
              <i
                className="like icon"
              />
              Like
            </span>,
            <span
              className="right floated star"
            >
              <i
                className="star icon"
              />
              Favorite
            </span>,
          ],
          "className": "extra content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="like icon"
                />,
                "Like",
              ],
              "className": "left floated like",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "like icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Like",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="star icon"
                />,
                "Favorite",
              ],
              "className": "right floated star",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "star icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Favorite",
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="content"
            onClick={[Function]}
          >
            <div
              className="header"
            />
            <div
              className="description"
            >
              
            </div>
          </div>,
          <div
            className="extra content"
          >
            <span
              className="left floated like"
            >
              <i
                className="like icon"
              />
              Like
            </span>
            <span
              className="right floated star"
            >
              <i
                className="star icon"
              />
              Favorite
            </span>
          </div>,
          undefined,
        ],
        "className": "training-card ui card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header"
              />,
              <div
                className="description"
              >
                
              </div>,
            ],
            "className": "content",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "header",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "description",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="left floated like"
              >
                <i
                  className="like icon"
                />
                Like
              </span>,
              <span
                className="right floated star"
              >
                <i
                  className="star icon"
                />
                Favorite
              </span>,
            ],
            "className": "extra content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="like icon"
                  />,
                  "Like",
                ],
                "className": "left floated like",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "like icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Like",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="star icon"
                  />,
                  "Favorite",
                ],
                "className": "right floated star",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "star icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Favorite",
              ],
              "type": "span",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
